scipp-imaging-2D.ipynb

```diff
@@ -196,246 +339,780 @@
+    "    xy_filtered = filter_func(yx)\n",
+    "    wavelength[k].data = sc.reshape(xy_filtered, dims=[\"wavelength\", \"spectrum\"], shape=(wavelength[k].shape[0],wavelength[k].shape[1]) )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Rebin to common bin width\n",
     "\n",
-    "zero_special_values(stitched[\"normalized\"].values)"
+    "As each detector has a different relative position to the sample we\n",
+    "need to rebin to a common set of wavelength bins for all"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "dm1d = sc.groupby(stitched[\"normalized\"], \"detector_mapping\", Dim.Row)\n",
-    "stitched[\"normalized_grpd\"] = dm1d.sum(Dim.Spectrum)"
+    "wav_x_vals = wavelength[\"sample\"].coords[\"wavelength\"]\n",
+    "start_wave = sc.min(wav_x_vals)\n",
+    "stop_wave = sc.max(wav_x_vals)\n",
+    "# sc.mean(wav_x_vals) throws an error, switch to using Numpy\n",
+    "bin_width = sc.Variable(unit=sc.units.angstrom, value=np.mean(wav_x_vals.values[:, 1:]-wav_x_vals.values[:, :-1]))\n",
+    "\n",
+    "# Arange does not include final bin so overshoot\n",
+    "stop_wave += bin_width\n",
+    "\n",
+    "wavelength_coords = sc.Variable([\"wavelength\"], values=np.arange(start_wave.value, stop_wave.value,\n",
+    "                                                                 bin_width.value))\n",
+    "rebinned_wavelength = sc.rebin(wavelength, \"wavelength\", wavelength_coords)\n",
+    "\n",
+    "\n",
+    "group_size = (324.0  / nx_target) * (324.0 / ny_target)\n",
+    "factor = group_size\n",
+    "rebinned_wavelength[\"sample\"] *= factor"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Sample - Sum Spectra\n",
+    "\n",
+    "Performs a weighted sum for all spectra in the sample "
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Plot the first group\n",
-    "sc.plot.plot(sc.sum(stitched[\"normalized_grpd\"], Dim.Row)) if do_plots else None"
+    "def weighted_sum(variable : sc.Variable, dim):\n",
+    "    variances = sc.Variable(dims=variable.dims, values=variable.variances)\n",
+    "    d = sc.Dataset()\n",
+    "    d['data'] = variable / variances\n",
+    "    d['norm'] = sc.reciprocal(variances)\n",
+    "\n",
+    "    d.masks['zero_error'] = sc.equal(variances, 0.0 * variances.unit)\n",
+    "    \n",
+    "    d = sc.sum(d, dim)\n",
+    "    sum = d['data']/d['norm']\n",
+    "    return sum\n",
```

Comment:
Just return without defining `sum`, which is a Python keyword and may (or may not?) break things?